lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp

```diff
@@ -758,6 +758,126 @@ void mlir::torch::onnx_c::populateDefaultDomainQtoZ(
             binder.op, resultType, operand, vAlpha, vScale, vInputScale);
         return success();
       });
+  patterns.onOp(
+      "ReduceL1", 1, [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+        Torch::ValueTensorType resultType;
+        int64_t keepDims, noop_with_empty_axes;
+        Value operand;
+        if (binder.tensorOperandAtIndex(operand, 0) ||
+            binder.tensorResultType(resultType) ||
+            binder.s64IntegerAttr(keepDims, "keepdims", 1) ||
+            binder.s64IntegerAttr(noop_with_empty_axes, "noop_with_empty_axes",
+                                  0))
+          return failure();
+
+        // create an intermediate dummy result type with same shape as operand
+        auto operandType = operand.getType().dyn_cast<Torch::ValueTensorType>();
+        Value data = rewriter.create<Torch::AtenAbsOp>(binder.getLoc(),
+                                                       operandType, operand);
+
+        // The code till the end is identical to ReduceSum Conversion Op below,
+        // as ReduceL1 is equivalent to ReduceSum(Abs(Operand)).
```

Comment:
If this is the case then the better approach is to extract this code and put it inside a function so that it can be reused for both these lowering and possible future lowerings.

Reply:
I extracted it into a common implementation, but it seems a little too complex. Is it OK if I push that for review?